### PR TITLE
Upgrade to from alpine 3.12 to 3.17

### PR DESCRIPTION
**Pull request by GwendalRaoul**

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-RUN ln -sf /usr/lib/libprotobuf.so.23 /usr/lib/libprotobuf.so
+RUN ln -sf /usr/lib/libprotobuf.so.32 /usr/lib/libprotobuf.so
 
 RUN python3 -m pip install wheel setuptools pkgconfig
 
 USER wirepas
 WORKDIR /home/wirepas
 
-# Protobuf checkout
+# Protobuf checkout and switch to tag that base image has (https://pkgs.alpinelinux.org/packages?name=protobuf&branch=v3.17&repo=&arch=&maintainer=)
```
Comment (pgaucherand) [inline docker/Dockerfile:14]: branch=3.17? sure?

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,2 @@
-# 3.12.2 version is used as it is
-# the version available in alpine 3.12 (for protobuf cpp version)
-protobuf>=3.12.2,<4.0
+# Minimum of 3.19 as protoc useed needed it
+protobuf>=3.19.2
```
Comment (pgaucherand) [inline requirements.txt:2]: are you sure it shouldn't be > 4.0? or does it matter? which version did you test?

**Review by pgaucherand — COMMENTED** [review]
Couple of comments from me

File: requirements.txt (hunk 2 above)
Comment (GwendalRaoul) [reply requirements.txt:2]: In this PR I updated the github action to use 3.20.3.
I cannot use a 4.x version as setup-protoc@v1 only support 3.x.x scheme

File: docker/Dockerfile (hunk 1 above)
Comment (pgaucherand) [reply docker/Dockerfile:14]: ok. so confusing all these 3.xxx

File: requirements.txt (hunk 2 above)
Comment (GwendalRaoul) [reply requirements.txt:2]: I tried to used 3.21.9 for protoc, but it is not an official release so action doesn't work.

But it seems to work well even if we use protoc 3.20.3 and 3.21.9 libprotobuf in alpine.

**Review by pgaucherand — APPROVED** [review]

**Pull request merged**